patches/server/0890-Nameable-Banner-API.patch

```diff
@@ -0,0 +1,36 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Thu, 7 Apr 2022 17:49:25 -0400
+Subject: [PATCH] Nameable Banner API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBanner.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBanner.java
+index 85f7ed3041befcc37729e9cd25723644600c7f62..edd1a637e966f2ed834d89cb4ee57419e4dfc4d8 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBanner.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBanner.java
+@@ -99,4 +99,25 @@ public class CraftBanner extends CraftBlockEntityState<BannerBlockEntity> implem
+         }
+         banner.itemPatterns = newPatterns;
+     }
++    // Paper start
++    @Override
++    public net.kyori.adventure.text.Component customName() {
++        return io.papermc.paper.adventure.PaperAdventure.asAdventure(this.getSnapshot().getCustomName());
++    }
++
++    @Override
++    public void customName(net.kyori.adventure.text.Component customName) {
++        this.getSnapshot().setCustomName(io.papermc.paper.adventure.PaperAdventure.asVanilla(customName));
++    }
++
++    @Override
++    public String getCustomName() {
++        return io.papermc.paper.adventure.PaperAdventure.LEGACY_SECTION_UXRC.serializeOrNull(this.customName());
++    }
++
++    @Override
++    public void setCustomName(String name) {
++       this.customName(io.papermc.paper.adventure.PaperAdventure.LEGACY_SECTION_UXRC.deserializeOrNull(name));
++    }
```

Comment:
How do we feel about adding new deprecated methods? I know its on the interface, but is there a better way to do this? Like should we make those methods on the Nameable interface be defaults that delegate to the component methods with the right serializer?

Reply:
I mean id say we can just make these default methods in the adventure patch yeah.